table-project/src/components/UserTable.tsx

```diff
@@ -29,63 +30,149 @@ const columns = [
 ];
   
 export default function UserTable() {
-  const { isLoading, isError, data } = useQuery({
-    queryKey: ["users"],
-    queryFn: () => fetchUsers(),
-    // Set staleTime to see that we get the "Server" output,
-    // can later switch tabs to trigger revalidation and see we
-    // get the "Client" output
-    staleTime: 10000,
-  });
+    const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
+        pageIndex: 0,
+        pageSize: 3,
+    })
 
-  const rerender = useReducer(() => ({}), {})[1];
+    const fetchDataOptions = {
+        pageIndex,
+        pageSize,
+    }
 
-  const options = {
-    data: data??[],
-    columns: columns,
-    getCoreRowModel: getCoreRowModel(),
-  }
+    const defaultData = useMemo(() => [], [])
 
-  const table = useReactTable(options);
+    const pagination = useMemo(
+        () => ({
+            pageIndex,
+            pageSize,
+        }),
+        [pageIndex, pageSize]
+    )
 
-  return (
-    <div>
-        {(isLoading || isError)? (<div>...</div>):(
-        <>
-        <table>
-            <thead>
-            {table.getHeaderGroups().map(headerGroup => (
-                <tr key={headerGroup.id}>
-                {headerGroup.headers.map(header => (
-                    <th key={header.id}>
-                    {header.isPlaceholder
-                        ? null
-                        : flexRender(
-                            header.column.columnDef.header,
-                            header.getContext()
-                        )}
-                    </th>
+    const rerender = useReducer(() => ({}), {})[1];
+
+    const { 
+        isLoading,
+        isError,
+        error,
+        data,
+        isFetching,
+        isPreviousData,
+    } = useQuery({
+        queryKey: ["data", fetchDataOptions],
+        queryFn: () => fetchUsers(fetchDataOptions),
+        keepPreviousData: true,
+    });
+
+    const options = {
+        data: data ?? defaultData,
+        columns: columns,
+        getCoreRowModel: getCoreRowModel(),
+        pageCount: 4,
+        state: {
+            pagination,
+        },
+        onPaginationChange: setPagination,
+        manualPagination: true,
+    }
```

Comment:
Page count is fixed here, but it would be pretty straightforward to bubble this up at the API implementation level.

We're using JSON placeholder API at the moment, so we just need to know that everything is working correctly.